compiler/main/analyzer/dead/dead_analyzer.h

```diff
@@ -6,7 +6,9 @@
  * are assumed to be live.
  */
 
-struct ST;
+#include "compiler/main/util/ctx.h"
+
 struct AST;
 
-void analyze_dead_code(struct ST* st, struct AST* ast);
+// @returns false on error
```

Comment:
I wonder if it would be worthwhile to return int here to differentiate between error cases but I given the existing logging inside the analyze function I doubt it.